### PR TITLE
Initial implementation of video model

### PR DESCRIPTION
**Pull request by i-like-robots**
If a CAPI v2 response is a video then this will intercept and use a different model that will fetch extra data from Brightcove.

### CODE REVIEW

File: server/models/article-v3.js [hunk 1]
```diff
@@ -68,6 +68,11 @@ function cleanAndSortStrings (items) {
 		});
 }
 
+function isVideo (articleV2) {
+	return articleV2.type === 'http://www.ft.com/ontology/content/MediaResource'
```
Comment (matthew-andrews) [inline server/models/article-v3.js]: worth pulling this a layer above this `article-v3.js` file — perhaps a `negotiation.js` file?

File: server/models/video.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		}
+
+		const renditions = brightcove.renditions.filter(rendition => {
+			return rendition.videoContainer === 'MP4';
+		});
+
+		return deepmerge(normalizeV2(capiV2), {
+			title: brightcove.name,
+			titles: [],
+			bodyXML: `<p>${brightcove.longDescription}</p>`,
```
Comment (matthew-andrews) [inline server/models/video.js]: i would suggest sticking a todo here as `bodyXML`'s are coming _very_ soon to videos in CAPI v2.

File: server/models/video.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+		'command=find_video_by_id',
+		'video_fields=id,name,renditions,longDescription,shortDescription,videoStillURL,customFields',
+		'token=' + process.env.BRIGHTCOVE_APIKEY,
+		'video_id=' + videoId
+	];
+
+	return fetch(`http://api.brightcove.com/services/library?${params.join('&')}`)
+		.then(response => {
+			return response.ok ? response.json() : response.text().then(text => {
+				return Promise.reject({ error: text, status: response.status });
```
Comment (matthew-andrews) [inline server/models/video.js]: clever…… but not very readable…… don't you think…… ?

File: server/models/article-v3.js (hunk 1 above)
Comment (i-like-robots) [inline server/models/article-v3.js]: Can do - though if UPP team can be convinced to do this for us then such a thing wouldn't be necessary.

**Comment by matthew-andrews** [comment]
couple of stylistic things and a request to add one `// todo`.  otherwise +1

File: server/models/video.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+			if (response.ok) {
+				return response.json();
+			}
+
+			return response.text().then(text => {
+				return Promise.reject({ error: text, status: response.status });
+			});
+		})
+		.catch(error => {
+			logger.warn({ event: 'BRIGHTCOVE_FETCH_FAIL', region: process.env.REGION, index: 'v3', uuid: videoId, status: error.status });
```
Comment (ironsidevsquincy) [inline server/models/video.js]: not logging `error.error`?

File: server/models/article-v3.js (hunk 1 above)
Comment (i-like-robots) [inline server/models/article-v3.js]: I confess, the more I look into this the more worried I am of getting stuck down a warren of refactoring so I've backed away for now =/

Comment (matthew-andrews) [inline server/models/article-v3.js]: it's ok, it's not necessary :-)

File: server/models/video.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+			};
+		}),
+		mainImage: {
+			title: `Video still for ${brightcove.name}`,
+			description: '',
+			url: brightcove.videoStillURL,
+			width: 640,
+			height: 360
+		},
+		_lastUpdatedDateTime: new Date().toISOString()
```
Comment (matthew-andrews) [inline server/models/video.js]: ah one final thing, I've been adding `url` property to content in advance of these sorts of things getting vanities.  Could you do the same for video?

https://github.com/Financial-Times/next-es-interface/blob/master/server/models/article/normalize-v2.js#L40

Comment (matthew-andrews) [inline server/models/video.js]: [ It just needs to be (for now) `/content/{uuid}` and I will add the magic for the vanities later ]

Comment (i-like-robots) [inline server/models/video.js]: https://github.com/Financial-Times/next-es-interface/pull/448/commits/6922838866b2daf39cf1ab0bebf25dd5d51a64fc

**Pull request merged**